optuna/notebooks/RAPIDS_xfeat_Optuna.ipynb

```diff
@@ -187,55 +285,76 @@
     "                           C=C,\n",
     "                           l1_ratio=l1_ratio,\n",
     "                           fit_intercept=fit_intercept,\n",
-    "                          selector=selector)\n",
+    "                           selector=selector)\n",
     "    return score\n",
     "\n",
-    "def feature_selection(df):\n",
+    "def feature_selection(df, experiment_name):\n",
     "    \"\"\"\n",
     "    Defines the Pipeline and performs the optuna opt\n",
     "    \"\"\"\n",
+    "    artifact_path = \"rapids-optuna-airline\"\n",
     "    selector = Pipeline(\n",
     "        [\n",
     "            SelectNumerical(),\n",
     "            KBestThresholdExplorer(ChiSquareKBest(target_col=\"ArrDelayBinary\")),\n",
     "        ]\n",
     "    )\n",
     "\n",
+    "    mlfcb = RAPIDSMLflowCallback(\n",
+    "        tracking_uri=MLFLOW_TRACKING_URI,\n",
+    "        experiment_name=experiment_name,\n",
+    "        metric_name='accuracy')\n",
     "    \n",
     "    study = optuna.create_study(direction=\"maximize\")\n",
-    "    study.optimize(partial(objective, df, selector), n_trials=N_TRIALS)\n",
-    "\n",
-    "    selector.from_trial(study.best_trial)\n",
-    "    selected_cols = selector.get_selected_cols()\n",
     "    \n",
-    "    return study, df[selected_cols]"
+    "\n",
+    "    mlflow.set_tracking_uri(MLFLOW_TRACKING_URI)\n",
```

Comment:
Do we need to set the URI and experiment here since the callback also registers?

Reply:
Yes, in fact if you don't, then it logs the top level model to the local file system, and the nested run to the sql database :) This bug caused me a great deal of confusion at one point.

It might make sense to leave it off of the callback, but then you'd always have to assume someone set the backend repo before running the callback, and that all callbacks would have to use the backend.

Reply:
Ahh okay. Sounds like a difficult one bug to catch!